src/client/main.tsx

```diff
@@ -24,22 +34,67 @@ export class Main extends React.Component<{}, ClientState> {
       console.log('connection!');
       this.startGame(initialData.gameState, initialData.playerId);
     });
-    this.setState({ stage: Stages.LOADING });
+    this.socket.on('new player', (numPlayers: number) => {
+      this.resetTimer();
+      this.updateCountdown(GAME_START_TIME);
+      this.setState({
+        stage: this.state.stage,
```

Comment:
Lets do
```js
this.setState({
  numPlayersInLobby: numPlayers,
  countdownTime: GAME_START_TIME,
} as ClientState);
```

instead, since theres a potential concurrency error by doing it this way.  `setState` is asynchronous, and so what could happen is that `this.state.stage` gets updated between this call to `setState` and the time it actually sets the state.  What would then happen is that when `setState` actually sets the state, it resets the updated value to its original value (the value when `setState` was first called)